File: 0155_Min Stack/MinStack.java
public class MinStack {

	private Stack<Integer> stack = new Stack<Integer>();
	private Stack<Integer> minStack = new Stack<Integer>();
	
    /** initialize your data structure here. */
    public MinStack() {
        
    }
    
    public void push(int x) {
        stack.push(x);
        if(minStack.size() ==0 || minStack.peek() >= x){
        	minStack.push(x);
        }
    }
    
    public void pop() {
    	if(stack.isEmpty()) return;
        int v = stack.pop();
        if(v == minStack.peek()){
        	minStack.pop();
        }
    }
    
    public int top() {
        return stack.peek();
    }
    
    public int getMin() {
        return minStack.peek();
    }
}

File: 0155_Min Stack/MinStack.py
class MinStack(object):

    def __init__(self):
        self.stack = list()
        self.minStack = list()
        
    def push(self, x):
        self.stack.append(x)
        if len(self.minStack) == 0 or self.minStack[len(self.minStack) - 1] >=x:
            self.minStack.append(x)
        

    def pop(self):
        if len(self.stack) ==0:
            return None
        
        v = self.stack.pop()
        if v == self.minStack[len(self.minStack) -1]:
            self.minStack.pop()

    def top(self):
        return self.stack[len(self.stack) -1]

    def getMin(self):
        return self.minStack[len(self.minStack)-1]